services/github/src/handlers.js

```diff
@@ -98,6 +100,14 @@ class Handlers {
     assert(!this.deprecatedResultStatusPq, 'Cannot setup twice!');
     assert(!this.deprecatedInitialStatusPq, 'Cannot setup twice!');
 
+    // This is a simple Queue client without scopes to use throughout the handlers for simple things
+    // Where scopes are needed, use this.queueClient.use({authorizedScopes: scopes}).blahblah
+    // (see this.createTasks for example)
+    this.queueClient = new taskcluster.Queue({
```

Comment:
I sorta don't like having this around in here since it is powerful and easy to make a mistake with. The comment says that it is for simple things where scopes aren't needed, but this client actually has _all_ of tc-github's scopes. I think we can just use the `authorizedScopes` client like before since fetching a task's status doesn't require any scopes.

Reply:
client with authorized scopes is created in `jobHandler` that has access to the scopes in `.taskcluster.yml`. Do you mean I should just create unscoped client every time it's needed as opposed to having it in the object?

Reply:
Also, I thought that when we create the scoped client, it has all the github scopes _and_ the scopes from `.taskcluster.yml`. Is that incorrect?

Reply:
Almost. When we use `authorizedScopes` it limits the set of scopes that the client can use to whatever the task is allowed based on the roles we assign it. Here's the scenario I'm scared of with the way this is built now:

Imagine in 6 months I come along and want to make a change to this file. I will look for the first queue client I have sitting around in here which very well could be `this.queueClient` and then I will use it to do some action that a task has requested. This is now a scope escalation because I didn't use the client with limited scopes. I would feel much safer if we just didn't ever generate a non-limited client in here.

Making a new client every time is a fine way to do it. I don't think it costs very much to do.